cmd/dagger/main.go

```diff
@@ -257,6 +262,15 @@ func checkForUpdates(ctx context.Context, w io.Writer) {
 	})
 }
 
+// checkCloudTokens triggers the Login flow if an invalid credentials file is detected.
+func checkCloudToken(ctx context.Context, w io.Writer) error {
+	_, err := auth.Token(ctx)
+	if v, ok := err.(*os.PathError); ok && v.Err == os.ErrNotExist {
```

Comment:
The comment says this will run for an invalid/malformed credentials file, but this seems to only check for the file not existing

Also I guess we need to be careful not to do anything if the user is intentionally not logged in (as in they don't use Cloud)

Reply:
> The comment says this will run for an invalid/malformed credentials file, but this seems to only check for the file not existing

agh! you're correct, forgot to commit some changes after testing errors. Should be ok in the follow-up commit.

Reply:
> Also I guess we need to be careful not to do anything if the user is intentionally not logged in (as in they don't use Cloud)

 yep, already checked this and we're not doing anything if the user has never logged in.